c/driver/postgresql/statement.cc

```diff
@@ -611,25 +605,26 @@ struct BindStream {
         return ADBC_STATUS_IO;
       }
 
-      if (PQputCopyEnd(conn, NULL) <= 0) {
-        SetError(error, "Error message returned by PQputCopyEnd: %s",
-                 PQerrorMessage(conn));
-        return ADBC_STATUS_IO;
-      }
+      if (rows_affected) *rows_affected += array->length;
```

Comment:
What we might want to do (certainly fine as a follow up) is send this data to the server at the end of each loop iteration using `PQputCopyData`:

https://www.postgresql.org/docs/current/libpq-copy.html#LIBPQ-COPY-SEND

Reply:
Sorry, what do you mean? We're still calling PQputCopyData for each batch, we're just not calling PQputCopyEnd until we've written *all* the batches

Reply:
(We could also be smarter here and flush to the server at a certain amount of data instead of each batch to smooth things out between large/small batches)

Reply:
Ah perfect. Ignore what I said then - I just missed it because I didn't see it in the diff